mobile/app/(app)/(drawer)/(tabs)/(observation)/form-details/[formId].tsx

```diff
@@ -138,14 +138,23 @@ const FormDetails = () => {
           answer = { ...answer, selectionValues: selectedAnswersTexts };
         }
 
+        // sort the notes by createdAt date in order to extract the text of the last added note
+        const sortedNotes =
+          notes && notes[q.id]
+            ? notes[q.id].slice().sort((a, b) => {
+                // added unary '+' operator to avoid ts error that the right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type
+                return +new Date(a.createdAt) - +new Date(b.createdAt);
+              })
+            : [];
+
         return {
           question: q.text[language],
           status: answers?.[q.id] ? QuestionStatus.ANSWERED : QuestionStatus.NOT_ANSWERED,
           answer,
           numberOfNotes: notes?.[q.id]?.length || 0,
           lastNoteText:
-            notes && notes?.[q.id]?.length > 0
-              ? notes?.[q.id][notes?.[q.id]?.length - 1].text
+            sortedNotes.length > 0
```

Comment:
If we need only the last note and the sortedNotes are not used to be displayed, we can keep only the last note instead of the full array, even if in this case the performance impact is minimal

Reply:
fixed in the last commit